File: src/me/ycdev/android/devtools/utils/ViewHelper.java
package me.ycdev.android.devtools.utils;

import android.content.Context;
import android.view.ViewGroup;
import android.view.ViewGroup.LayoutParams;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

public class ViewHelper {
    public static void addTextView(ViewGroup holder, String itemValue) {
        TextView itemView = new TextView(holder.getContext());
        itemView.setLayoutParams(new LayoutParams(
                LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT));
        itemView.setText(itemValue);
        holder.addView(itemView);
    }

    public static void addLineView(ViewGroup holder, int color) {
        Context context = holder.getContext();

        ImageView lineView = new ImageView(context);
        lineView.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, 1));
        lineView.setBackgroundColor(color);

        LinearLayout marginView = new LinearLayout(context);
        marginView.setPadding(0, 6, 0, 6);
        marginView.setLayoutParams(new LayoutParams(
                LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT));
        marginView.addView(lineView);

        holder.addView(marginView);
    }

    public static void addTextView(ViewGroup holder, String itemName, String itemValue) {
        addTextView(holder, itemName + ": " + itemValue);
    }

    public static void addTextView(ViewGroup holder, String itemName, boolean itemValue) {
        addTextView(holder, itemName, String.valueOf(itemValue));
    }

    public static void addTextView(ViewGroup holder, String itemName, int itemValue) {
        addTextView(holder, itemName, String.valueOf(itemValue));
    }

    public static void addTextView(ViewGroup holder, String itemName, double itemValue) {
        addTextView(holder, itemName, String.valueOf(itemValue));
    }

    public static void addTimeTextView(ViewGroup holder, String itemName, long macroSeconds) {
        int seconds = (int) (macroSeconds / (1000 * 1000));
        String timeStr = FormatHelper.formatElapsedTime(holder.getContext(), seconds);
        addTextView(holder, itemName, timeStr);
    }

    public static void addBytesTextView(ViewGroup holder, String itemName, long bytes) {
        String bytesStr = FormatHelper.formatBytes(holder.getContext(), bytes);
        addTextView(holder, itemName, bytesStr);
    }

}

File: app/src/main/java/me/ycdev/android/devtools/utils/ViewHelper.kt
package me.ycdev.android.devtools.utils

import android.view.ViewGroup
import android.view.ViewGroup.LayoutParams
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

object ViewHelper {
    fun addTextView(holder: ViewGroup, itemValue: String?) {
        val itemView = TextView(holder.context)
        itemView.layoutParams = LayoutParams(
            LayoutParams.WRAP_CONTENT,
            LayoutParams.WRAP_CONTENT
        )
        itemView.text = itemValue
        holder.addView(itemView)
    }

    fun addLineView(holder: ViewGroup, color: Int) {
        val context = holder.context
        val lineView = ImageView(context)
        lineView.layoutParams = LayoutParams(
            LayoutParams.MATCH_PARENT,
            1
        )
        lineView.setBackgroundColor(color)
        val marginView = LinearLayout(context)
        marginView.setPadding(0, 6, 0, 6)
        marginView.layoutParams = LayoutParams(
            LayoutParams.MATCH_PARENT,
            LayoutParams.WRAP_CONTENT
        )
        marginView.addView(lineView)
        holder.addView(marginView)
    }

    fun addTextView(
        holder: ViewGroup,
        itemName: String,
        itemValue: String?
    ) {
        addTextView(holder, "$itemName: $itemValue")
    }

    fun addTextView(
        holder: ViewGroup,
        itemName: String,
        itemValue: Boolean
    ) {
        addTextView(holder, itemName, itemValue.toString())
    }

    fun addTextView(
        holder: ViewGroup,
        itemName: String,
        itemValue: Int
    ) {
        addTextView(holder, itemName, itemValue.toString())
    }

    fun addTextView(
        holder: ViewGroup,
        itemName: String,
        itemValue: Double
    ) {
        addTextView(holder, itemName, itemValue.toString())
    }

    fun addTimeTextView(
        holder: ViewGroup,
        itemName: String,
        macroSeconds: Long
    ) {
        val seconds = (macroSeconds / (1000 * 1000)).toInt()
        val timeStr = FormatHelper.formatElapsedTime(holder.context, seconds)
        addTextView(holder, itemName, timeStr)
    }

    fun addBytesTextView(
        holder: ViewGroup,
        itemName: String,
        bytes: Long
    ) {
        val bytesStr = FormatHelper.formatBytes(holder.context, bytes)
        addTextView(holder, itemName, bytesStr)
    }
}
